cmd/helm/testdata/output/list-all.txt

```diff
@@ -5,5 +5,6 @@ groot      	default  	1       	2016-01-16 00:00:01 +0000 UTC	uninstalled    	chi
 hummingbird	default  	1       	2016-01-16 00:00:03 +0000 UTC	deployed       	chickadee-1.0.0	0.0.1      
 iguana     	default  	2       	2016-01-16 00:00:04 +0000 UTC	deployed       	chickadee-1.0.0	0.0.1      
 rocket     	default  	1       	2016-01-16 00:00:02 +0000 UTC	failed         	chickadee-1.0.0	0.0.1      
+starlord   	default  	1       	2016-01-16 00:00:01 +0000 UTC	superseded     	chickadee-1.0.0	0.0.1      
```

Comment:
The goal with `helm list --all` is to show the latest revision of every release, regardless of what state the latest release is in. So this does appear like it would break backwards compatibility, as it not only shows the latest revision but EVERY revision in the ledger.

Can you look into refactoring the state mask while retaining prior behaviour?

Reply:
Thanks for clarifying, @bacongobbler! The expected behavior of `helm list --all` wasn't clear to me.

Reply:
Getting a historical list of a particular release is precisely what I thought `pkg/action.List` was used for. I can see now that I misunderstood.

I think that part of my confusion comes from learning helm semantics from the SDK (i.e. `pkg/action`), rather than the CLI. I will close this PR, and the original issue, and instead open a PR adding some comments to `pkg/action.List`, so that the purpose of `action.List` and `action.History` is clear.

Reply:
Filed #9139 with comments. Closing this PR.